test/test_transformers.py

```diff
@@ -1436,6 +1435,28 @@ def test_flash_backward_failure_sm86plus(self, head_dim: int):
             self.assertRaises(RuntimeError, lambda: torch.nn.functional.scaled_dot_product_attention(
                 q, k, v, None, 0.0, False))
 
+    @unittest.skipIf(not PLATFORM_SUPPORTS_FUSED_SDPA or not isSM90Device,
+                     "Does not support fused SDPA or not SM90 hardware")
+    @parametrize("head_dim", [32, 64, 72, 96, 128])
+    def test_flash_backward_success_sm90(self, head_dim: int):
+        device = 'cuda'
+        dtype = torch.float16
+        make_tensor = partial(self.rand_tensor, type="dense", device=device, dtype=dtype)
+        size = (2, 2, 4, head_dim)
+        q, k, v = make_tensor(size, requires_grad=True), make_tensor(
+            size, requires_grad=True), make_tensor(size, requires_grad=True)
+
+        with sdp_kernel(enable_mem_efficient=False, enable_flash=False, enable_math=True):
+            math_ref = torch.nn.functional.scaled_dot_product_attention(q, k, v, None, 0.0, False)
+            math_ref.mean().backward()
+
+        with sdp_kernel(enable_mem_efficient=False, enable_flash=True, enable_math=False):
+            flash_ref = torch.nn.functional.scaled_dot_product_attention(q, k, v, None, 0.0, False)
+            self.assertEqual(math_ref, flash_ref, atol=1e-3, rtol=1e-3)
+
+            flash_ref.mean().backward()
+            self.assertEqual(math_ref.grad, flash_ref.grad, atol=1e-3, rtol=1e-3)
+
```

Comment:
Why do we need a specific test for sm_90? Wouldn't existing tests that run on sm_80 would also run on sm_90?

Reply:
From what I understand, the existing test that tests the `with sdp_kernel(enable_mem_efficient=False, enable_flash=True, enable_math=False)` path as it currently is (https://github.com/pytorch/pytorch/blob/main/test/test_transformers.py#L1415) would expect a RuntimeError for all SM86+ with headdim > 64. In particular, this would expect a RuntimeError for SM90 with headdim > 64, but this PR enables SM90 with headdim > 64 so it would work successfully, thus failing the assert. So the new test is SM90 specific and also verifies that the backward grads are numerically accurate.

Reply:
You should expand the pamertrization here: https://github.com/pytorch/pytorch/blob/f274c4ecf68d989f921efb685cfb3177cf2e6220/test/test_transformers.py#L1730 
and also add a similiar failing assert like this: https://github.com/pytorch/pytorch/blob/f274c4ecf68d989f921efb685cfb3177cf2e6220/test/test_transformers.py#L1673

Reply:
Thanks for the suggestion @drisspg, I updated the test_transformer.py to remove this new test and follow your suggestion to reuse the existing `test_flash_attention_vs_math_ref_grads` with added parameterization and assert for sm86 and sm89.